### ISSUE TITLE
synda remove removed whole /srv tree

### ISSUE THREAD

**Issue by Gringo5000** [Prodiguer/synda#3]
There may be some user error here, but I tried to install a sample dataset. It created the directory tree in /srv/synda/sdt/cmip5/... but failed to transfer the file (error 27). I later tried the transfer again, but it said the files were already installed:
```
[root@aimsdtn5:sdt]# sudo synda install cmip5.output1.MPI-M.MPI-ESM-LR.decadal1995.mon.land.Lmon.r2i1p1.v20120529 baresoilFrac
http://esgf-data1.ceda.ac.uk/thredds/fileServer/esg_dataroot/cmip5/output1/MPI-M/MPI-ESM-LR/decadal1995/mon/land/Lmon/r2i1p1/v20120529/baresoilFrac/baresoilFrac_Lmon_MPI-ESM-LR_decadal1995_r2i1p1_199601-200512.nc
http://esgf-data1.ceda.ac.uk/thredds/fileServer/esg_dataroot/cmip5/output1/MPI-M/MPI-ESM-LR/decadal1995/mon/land/Lmon/r2i1p1/v20120529/baresoilFrac/baresoilFrac_Lmon_MPI-ESM-LR_decadal1995_r2i1p1_199601-200512.nc

Nothing to install (files already installed).
```

The queue still showed the error state:
```
[root@aimsdtn5:sdt]# synda queue
                                                                                                    

status      count  size
error           1  8.9 MB
```
I then tried a 'synda remove' to see if that would clear the installed state. I removed it, then tried re-installing the dataset.  That command showed some errors and a subsequent attempt to install the dataset complained the directory tree was missing, which 'ls' confirmed.
```
[root@aimsdtn5:sdt]# synda remove cmip5.output1.MPI-M.MPI-ESM-LR.decadal1995.mon.land.Lmon.r2i1p1.v20120529 baresoilFrac
http://esgf-data1.ceda.ac.uk/thredds/fileServer/esg_dataroot/cmip5/output1/MPI-M/MPI-ESM-LR/decadal1995/mon/land/Lmon/r2i1p1/v20120529/baresoilFrac/baresoilFrac_Lmon_MPI-ESM-LR_decadal1995_r2i1p1_199601-200512.nc
http://esgf-data1.ceda.ac.uk/thredds/fileServer/esg_dataroot/cmip5/output1/MPI-M/MPI-ESM-LR/decadal1995/mon/land/Lmon/r2i1p1/v20120529/baresoilFrac/baresoilFrac_Lmon_MPI-ESM-LR_decadal1995_r2i1p1_199601-200512.nc
1 file(s) will be removed.
After this operation, 0 Bytes of disk space will be freed.
Do you want to continue? [y/N] y
1 file(s) removed

*** Error occured at 2016-02-19 14:09:48.902619 ***


==================
*   Error code   *
==================

SDOPERAT-001


=====================
*   Error message   *
=====================

Error occurs during tree cleanup

[root@aimsdtn5:sdt]# synda install cmip5.output1.MPI-M.MPI-ESM-LR.decadal1995.mon.land.Lmon.r2i1p1.v20120529 baresoilFrac
Traceback (most recent call last):
  File "/usr/bin/synda", line 25, in <module>
    import sdapp
  File "/usr/share/python/synda/sdt/bin/sdapp.py", line 21, in <module>
    import sdconfig
  File "/usr/share/python/synda/sdt/bin/sdconfig.py", line 133, in <module>
    check_path(data_folder)
  File "/usr/share/python/synda/sdt/bin/sdconfig.py", line 50, in check_path
    raise SDException("SDATYPES-101","Path not found (%s)"%path)
sdexception.SDException: code=SDATYPES-101,message=Path not found (/srv/synda/sdt)
```

**Comment by jripsl** [comment]
Hi,

> It created the directory tree in /srv/synda/sdt/cmip5/... 
> but failed to transfer the file (error 27). 
>
This error is related to permission (e.g. missing cmip5-research role).

> I later tried the transfer again, but it said the files were already installed:
>
Here "already installed" means the file is already marked for download.

I changed both error messages, so it should be more  in the next release (3.3).

> I then tried a 'synda remove' to see if that would clear the installed state. 
> I removed it, then tried re-installing the dataset. 
> That command showed some errors
>
Bug in cleanup script which affect RPM and DEB package based installation.
It's fixed in 3.3 release.
To fix the bug in 3.2 release, run both commands below:
```
mkdir -p /srv/synda/sdt
sudo wget http://dods.ipsl.jussieu.fr/jripsl/synda/patches/3.2/sdcleanup_tree.sh -O /usr/share/python/synda/sdt/bin/sdcleanup_tree.sh
```

Thanks for your feedback

**Comment by jripsl** [comment]
Hi,

> It created the directory tree in /srv/synda/sdt/cmip5/... 
> but failed to transfer the file (error 27). 
>
This error is related to permission (e.g. missing cmip5-research role)

> I later tried the transfer again, but it said the files were already installed:
>
"already installed" means the files are already marked for download.

I changed both error messages, so it should be more explicit in the next release (3.3).

> I then tried a 'synda remove' to see if that would clear the installed state. 
> I removed it, then tried re-installing the dataset. 
> That command showed some errors
>
This is caused by a bug in cleanup script which affect RPM and DEB installation.
It's fixed in 3.3 release.
To fix the bug in 3.2 release, run both commands below:
```
mkdir -p /srv/synda/sdt
sudo wget http://dods.ipsl.jussieu.fr/jripsl/synda/patches/3.2/sdcleanup_tree.sh -O /usr/share/python/synda/sdt/bin/sdcleanup_tree.sh
```

Thanks for your feedback

**Comment by Gringo5000** [comment]
Thanks for the quick response.
